### ISSUE TITLE
cdc scale-out（2 to 3）OOM

### ISSUE THREAD

**Issue by buggithubs** [pingcap/tiflow#4844]
### What did you do?

1. Synchronized data
2. There is a delay
3. Expansion nodes

<img width="1228" alt="image" src="https://user-images.githubusercontent.com/11624213/157636685-14bb09e1-9c33-4a46-9448-a3e9f3a27bac.png">

### What did you see instead?

![0NgXBifmGr](https://user-images.githubusercontent.com/11624213/157636343-3192aa17-b998-4ba1-8af3-ff46e89ea48e.png)
![aUhhbtoqAF](https://user-images.githubusercontent.com/11624213/157636367-bd573620-4132-49d8-b22a-532190d9167f.png)

### Versions of the cluster

Upstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):

```console
(paste TiDB cluster version here)
TiDB version 5.4.0
TiKV version 5.4.0
CDC version 5.4.0
```

**Comment by Tammyxia** [comment]
Before cdc scale-out, cdc works well:
![image](https://user-images.githubusercontent.com/78345569/157639389-bee26103-696a-4310-a289-7fb7419a11b0.png)
table count = 960,  all tables have been created in downstream before cdc scale-out.

**Comment by overvenus** [comment]
After some offline tests, we managed to run TiCDC without OOM by adding following configuration

```toml
debug.enable-db-sorter = true
per-table-memory-quota = 548576 # 512KB
``` 

Summary of the OOM issue.

1. TiCDC runs on SAS HDD, the disk is slow. Upstream workload is batch insert, TiCDC takes long time to re-initialize after scale-out.
2. There are too many tables, TiCDC runs out memory easily with default configuration `per-table-memory-quota= 10485760` (10MB), it takes about 9.6GB for flow control.

Also we identified another issue which has been solved in master branch.

3. The default block cache size of leveldb sorter (4GB) is too big for this scenario, it should adjust automatically base on the system's total memory. https://github.com/pingcap/tiflow/pull/4632/files#diff-4f24fb50f8768cd676f96ed676da601c83670784c3af3fac055361e7b035efd9R173


<img width="609" alt="image" src="https://user-images.githubusercontent.com/2150711/158107721-c4828853-9f4e-4906-b5b9-1f42ccc84d7c.png">
<img width="611" alt="image" src="https://user-images.githubusercontent.com/2150711/158108857-a18568a6-d926-4916-832d-d30f88e887a4.png">
<img width="458" alt="image" src="https://user-images.githubusercontent.com/2150711/158108902-55737f0e-266d-49a8-8222-98cdb1213bd9.png">
